lms/djangoapps/discussion/rest_api/permissions.py

```diff
@@ -62,38 +62,39 @@ def get_editable_fields(cc_content, context):
     # For closed thread:
     # no edits, except 'abuse_flagged' and 'read' are allowed for thread
     # no edits, except 'abuse_flagged' is allowed for comment
-    ret = {"abuse_flagged"}
-    if cc_content["type"] == "thread" and cc_content["closed"]:
-        ret |= {"read"}
-        return ret
-    if cc_content["type"] == "comment" and context["thread"]["closed"]:
-        return ret
-
-    # Shared fields
-    ret |= {"voted"}
-    if _is_author_or_privileged(cc_content, context):
-        ret |= {"raw_body"}
-
-    # Thread fields
-    if cc_content["type"] == "thread":
-        ret |= {"following", "read"}
-        if _is_author_or_privileged(cc_content, context):
-            ret |= {"topic_id", "type", "title"}
-        if context["is_requester_privileged"] and context["discussion_division_enabled"]:
-            ret |= {"group_id"}
-
-    # Comment fields
-    if (
-            cc_content["type"] == "comment" and (
-                context["is_requester_privileged"] or (
-                    _is_author(context["thread"], context) and
-                    context["thread"]["thread_type"] == "question"
-                )
-            )
-    ):
-        ret |= {"endorsed"}
-
-    return ret
+    is_thread = cc_content["type"] == "thread"
+    is_comment = cc_content["type"] == "comment"
+    is_privileged = context["is_requester_privileged"]
+    is_thread_closed = cc_content["closed"] if is_thread else context["thread"]["closed"]
+
+    # Map each field to the condition in which it's editable.
+    editable_fields = {
+        "abuse_flagged": True,
+        "closed": is_thread and is_privileged,
+        "pinned": is_thread and is_privileged,
+        "read": is_thread,
+    }
+
+    if is_thread_closed:
+        # Return only true entries
+        return {field for field, is_editable in editable_fields.items() if is_editable}
```

Comment:
Is it worth extracting this into a function?

Reply:
As an alternative:
```python
def get_editable_fields(cc_content, context):
    editable_fields_map = _get_editable_fields_map(cc_content, context)
    return {field for field, is_editable in editable_fields_map.items() if is_editable}
```

Reply:
It's a very small code fragment, and I don't think either case solves much. I don't see this pattern any many other places in the platform. If it did exist I think it would make sense to consolidate the logic.